src/metadata_validator.py

```diff
@@ -125,6 +127,27 @@ def validate(self, submission_id, scope):
         self.submission = submission
         datacommon = submission.get(DATA_COMMON_NAME)
         self.datacommon = datacommon
+        # get study name and program name(s) from submission and/or study for name validation required in CRDCDH-2431
+        study_id = submission.get(STUDY_ID)
+        if not study_id:
+            msg = f'Invalid submission, no study id found, {submission_id}!'
+            self.log.error(msg)
+            return FAILED
+        study = self.mongo_dao.find_study_by_id(study_id)
+        if not study:
+            msg = f'Invalid submission, no study found, {submission_id}!'
+            self.log.error(msg)
+            return FAILED
+        self.study_name = study.get("studyName")
+        program = submission.get("organization") 
```

Comment:
"organization" property in Submission should be removed. Please find program via study_id only.